### DISCUSSION TITLE
v7.8.0

### DISCUSSION THREAD

**Discussion by jeffsmale90** [ConsenSys-archive/ganache#4352] [Releases]
<a id="user-content-v7.8.0-top"></a>
<h4>
  <p align="center">
		<code>&nbsp;<a href="#user-content-v7.8.0-new-features">New&nbsp;Features</a>&nbsp;</code>
  <img height="36" width="0" src="https://raw.githubusercontent.com/davidmurdoch/px/master/1px.gif">
		<code>&nbsp;<a href="#user-content-v7.8.0-changelog">Changelog</a>&nbsp;</code>
  <img height="36" width="0" src="https://raw.githubusercontent.com/davidmurdoch/px/master/1px.gif">
		<code>&nbsp;<a href="#user-content-v7.8.0-known-issues">Known&nbsp;Issues</a>&nbsp;</code>
  <img height="36" width="0" src="https://raw.githubusercontent.com/davidmurdoch/px/master/1px.gif">
		<code>&nbsp;<a href="#user-content-v7.8.0-future-plans">Future&nbsp;Plans</a>&nbsp;</code>
  <img height="36" width="0" src="https://raw.githubusercontent.com/davidmurdoch/px/master/1px.gif">
  </p>
</h4>

---

Introducing the newest addition to our lineup - Ganache v7.8.0! This release comes with support for the highly anticipated Shanghai hardfork.

If you have some time, we encourage you to browse our [issues](https://github.com/trufflesuite/ganache/issues) to find anything you'd like implemented/fixed sooner. Give them a +1 and we'll use this community feedback to help prioritize what we work on! Or better yet, [open a new issue](https://github.com/trufflesuite/ganache/issues/new), or [open a PR](https://github.com/trufflesuite/ganache/compare) to fix an existing issue.

We've changed 84 files across 1 merged pull requests, tallying 6,243 additions and 4,642 deletions, since our last release.

<a id="user-content-v7.8.0-new-features" ></a>

---

# <p align="center"><a href="#user-content-v7.8.0-new-features"><img alt="New Features" width="auto" src="https://raw.githubusercontent.com/trufflesuite/ganache/release-notes-assets/title-images/new-features.svg"></a></p>

### <a id="user-content-v7.8.0-new-features-0"></a>feat: add support for the `shanghai` hardfork (#4272)

* [EIP-3651: Warm COINBASE](https://eips.ethereum.org/EIPS/eip-3651)
* [EIP-3855: PUSH0 instruction](https://eips.ethereum.org/EIPS/eip-3855)
* [EIP-4895: Beacon chain push withdrawals as operations](https://eips.ethereum.org/EIPS/eip-4895)
* [EIP-6049: Deprecate SELFDESTRUCT](https://eips.ethereum.org/EIPS/eip-6049)
* [EIP-3860: Limit and meter initcode](https://eips.ethereum.org/EIPS/eip-3860)

We've added a new option `allowUnlimitedInitCodeSize` to disable EIP-3860 (Limit and meter initcode) to continue to allow for uploading very large contracts. In most cases it will need to be combined with the `allowUnlimitedContractSize` option. 
  
This also fixes two bugs:

* block `size` was not computed correctly (persisted databases will be updated and migrated automatically)
  * the database migration may add a one-time delay to startup when using a persistent database (`--database.dbPath`). This should be insignificant for the majority of cases. The migration was benchmarked with 180,000 blocks, and completed in 6.4 seconds.
* internal "intrinsic gas" checks now correctly include `accessList` fees, if there are any.

<a id="user-content-v7.8.0-changelog" ></a>

---

# <p align="center"><a href="#user-content-v7.8.0-changelog"><img alt="Changelog" width="auto" src="https://raw.githubusercontent.com/trufflesuite/ganache/release-notes-assets/title-images/changelog.svg"></a></p>

- #4272 feat: support `shanghai` hardfork (@davidmurdoch)

<p align="right"><sup><a href="#user-content-v7.8.0-top">back to top</a></sup></p>
        

<a id="user-content-v7.8.0-known-issues" ></a>

---

# <p align="center"><a href="#user-content-v7.8.0-known-issues"><img alt="Known Issues" width="auto" src="https://raw.githubusercontent.com/trufflesuite/ganache/release-notes-assets/title-images/known-issues.svg"></a></p>

### Top Priority:
 - interactive documentation's `debug_storageRangeAt` doesn't work ([#3203](https://github.com/trufflesuite/ganache/issues/3203))
 - Add `eth_createAccessList` RPC method ([#1056](https://github.com/trufflesuite/ganache/issues/1056))
### Coming Soon™:
 - Implications failed:  fork.headers -> url ([#2627](https://github.com/trufflesuite/ganache/issues/2627))
 - In Geth chain-mode, logic to accept/reject transactions based on gas price/limit should match Geth ([#2176](https://github.com/trufflesuite/ganache/issues/2176))
 - `evm_mine` and `miner_start` don't respect --mode.instamine=eager ([#2029](https://github.com/trufflesuite/ganache/issues/2029))
 - `evm_setAccount*` is race-conditiony ([#1646](https://github.com/trufflesuite/ganache/issues/1646))
 - `@ganache/filecoin@alpha` doesn't work with `ganache@alpha` ([#1150](https://github.com/trufflesuite/ganache/issues/1150))
 - Launching ganache with fork is throwing revert errors when communicating with 3rd party contracts ([#956](https://github.com/trufflesuite/ganache/issues/956))
 - Build a real pending block! ([#772](https://github.com/trufflesuite/ganache/issues/772))
 - VM Exception when interfacing with Kyber contract ([#606](https://github.com/trufflesuite/ganache/issues/606))
 - After calling `evm_mine`, `eth_getLogs` returns same logs for all blocks ([#533](https://github.com/trufflesuite/ganache/issues/533))
 - personal_unlockAccount works with any password ([#165](https://github.com/trufflesuite/ganache/issues/165))
 - --db Option Requires Same Mnemonic and Network ID ([#1030](https://github.com/trufflesuite/ganache/issues/1030))

<p align="right"><sup><a href="#user-content-v7.8.0-top">back to top</a></sup></p>
        

<a id="user-content-v7.8.0-future-plans" ></a>

---

# <p align="center"><a href="#user-content-v7.8.0-future-plans"><img alt="Future Plans" width="auto" src="https://raw.githubusercontent.com/trufflesuite/ganache/release-notes-assets/title-images/future-plans.svg"></a></p>

### Top Priority:
 - Accept a genesis.json file ([#1042](https://github.com/trufflesuite/ganache/issues/1042))
### Coming Soon™:
 - Switch to esbuild to make build times faster/reasonable ([#1555](https://github.com/trufflesuite/ganache/issues/1555))
 - fork specific block & specific index ([#952](https://github.com/trufflesuite/ganache/issues/952))
 - Allow to sync forked chain to the latest block ([#643](https://github.com/trufflesuite/ganache/issues/643))
 - Implement a streaming trace capability ([#381](https://github.com/trufflesuite/ganache/issues/381))
 - Improve log performance when forking ([#145](https://github.com/trufflesuite/ganache/issues/145))
 - Log contract events ([#45](https://github.com/trufflesuite/ganache/issues/45))

<p align="right"><sup><a href="#user-content-v7.8.0-top">back to top</a></sup></p>
        

[Open new issues](https://github.com/trufflesuite/ganache/issues/new?milestone=7.0.0) to influence what gets implemented and prioritized.

---

<p align="center">
  💖 The Truffle Team
</p>

<hr /><em>This discussion was created from the release <a href='https://github.com/trufflesuite/ganache/releases/tag/v7.8.0'>v7.8.0</a>.</em>

**Comment by EvilJordan** [comment]
I see that it's listed in the `new features` section, but could you/someone explain the right way to simulate a withdrawal “event”?

**Reply by davidmurdoch** [reply]
There isn't a way right now. I'm super interested in your use case and how we can build something useful for withdrawals here though. Do you mind opening a new discussion to talk more about what you'd like to do with withdrawals?

**Reply by EvilJordan** [reply]
https://github.com/trufflesuite/ganache/issues/4371

Here ya go!
